avocado/utils/archive.py

```diff
@@ -39,6 +40,37 @@
         LZMA_CAPABLE = False
 
 
+#: The first two bytes that all gzip files start with
+GZIP_MAGIC = b'\037\213'
+
+
+def is_gzip_file(path):
+    """
+    Checks if file given by path has contents that suggests gzip file
+    """
+    with open(path, 'rb') as gzip_file:
+        return gzip_file.read(len(GZIP_MAGIC)) == GZIP_MAGIC
+
+
+def gzip_uncompress(path, output_path):
+    """
+    Uncompress a gzipped file at path, to either a file or dir at output_path
+    """
+    with gzip.GzipFile(filename=path, mode='rb') as input_file:
+        if os.path.isdir(output_path):
+            basename = os.path.basename(path)
+            if basename.endswith('.gz'):
+                basename = basename[:-3]
```

Comment:
So, if it's a `myfile.tar.gz` it's going to be uncompressed in `output_path/myfile.tar`. Right? What about get the name until the first occurence of a dot?